providers/src/airflow/providers/dbt/cloud/utils/openlineage.py

```diff
@@ -142,7 +142,7 @@ async def get_artifacts_for_steps(steps, artifacts):
         parent_run_id = OpenLineageAdapter.build_task_instance_run_id(
             dag_id=task_instance.dag_id,
             task_id=operator.task_id,
-            execution_date=task_instance.execution_date,
+            logical_date=task_instance.logical_date,
```

Comment:
I think this needs a version check too?

Reply:
I think this change is not needed for this PR right as per https://github.com/apache/airflow/pull/43902#discussion_r1839379068?

Reply:
The argument is not, but `task_instance.logical_date` does not work on 2.